RoomTag implementation and functionality

Tervoc opened the pull request:
room tags are generated by a member of a room and can be used to invite other roomee users to their room.
fixes other minor issues and functions in classes that were incorrectly implemented.
Fixes #2

Commit:
RoomTag implementation and functionality 

room tags are generated by a member of a room and can be used to invite other roomee users to their room.

Commit:
Authored comments

added authored by, creation date, and notes comments for all CS files

mattpadgett commented on roomee-api/roomee-api/Controllers/RoomController.cs:
```diff
@@ -19,7 +24,7 @@ namespace roomee_api.Controllers {
 	public class RoomController : ControllerBase {
 		[HttpGet]
 		public IActionResult GetRoom([FromQuery][Required] string type, [FromQuery][Required] string identifier) {
```
This function takes in a type and an identifier. As far as I can see, the Room is only looked up by id and nothing else. In this case, you would only need to take in a single integer for the id as it is the only method for getting a Room. Then, the [HttpGet] would become [Http("{id}")].

mattpadgett commented on roomee-api/roomee-api/Controllers/RoomController.cs:
```diff
@@ -40,48 +45,37 @@ public class RoomController : ControllerBase {
-				if (rows == 0) {
-					return Problem("error creating");
-				}
+			Dictionary<string, string> userVals = Authentication.ReadToken(token);
```
Should be using Authentication.IsTokenValid method to verify that the token is valid. The ReadToken method does not check to determine if the token is valid.

mattpadgett commented on roomee-api/roomee-api/Controllers/RoomTagController.cs:
```diff
@@ -0,0 +1,129 @@
+	[ApiController]
+	public class RoomTagController : ControllerBase {
+		[HttpGet]
+		public IActionResult GetRoomTag([FromQuery][Required] string type, [FromQuery][Required] string identifier) {
```
See the previous comment about the type/identifier issue. Also, this takes in a RoomId in the RoomTag controller. If you meant RoomTagId then that's fine, but if you actually do mean RoomId then this doesn't work because a room could have multiple room tags.

mattpadgett commented on roomee-api/roomee-api/Controllers/RoomTagController.cs:
```diff
@@ -0,0 +1,129 @@
+		}
+
+		[HttpPost("generate")]
+		public IActionResult GenerateNewTag ([FromQuery][Required] int roomId) {
```
Don't forget to check their token. Also, shouldn't they only be able to create a room tag if they are a member of that room? You would need to pull their userId from the token in order to check this.

mattpadgett commented on roomee-api/roomee-api/Controllers/RoomTagController.cs:
```diff
@@ -0,0 +1,129 @@
+				return Problem("room tag cannot be empty");
+			}
+
+			Dictionary<string, string> userVals = Authentication.ReadToken(token);
```
Validate the token, please.

mattpadgett commented on roomee-api/roomee-api/Controllers/RoomTagController.cs:
```diff
@@ -0,0 +1,129 @@
+
+			Dictionary<string, string> userVals = Authentication.ReadToken(token);
+
+			if (!userVals.ContainsKey("userId") || userVals["userId"] == string.Empty || userVals["userId"] == null) {
```
Don't worry about this. All valid tokens will have it..

mattpadgett commented on roomee-api/roomee-api/Controllers/RoomTagController.cs:
```diff
@@ -0,0 +1,129 @@
+		}
+
+		[HttpPatch("{id}")]
+		public IActionResult UpdateRoom([FromRoute] int id, [FromBody] Dictionary<string, string> patch) {
```
Don't forget to validate the token.

mattpadgett commented on roomee-api/roomee-api/Controllers/UserController.cs:
```diff
@@ -42,10 +47,10 @@ public class UserController : ControllerBase {
 		}
 
-		[HttpPost]
+		[HttpPost("create")]
```
This doesn't need to be named create. It is inferred from the post request that you will be creating something. The same goes for the previous file's generate name.

mattpadgett commented on roomee-api/roomee-api/Models/Announcement.cs:
```diff
@@ -0,0 +1,73 @@
+		[JsonProperty(PropertyName = "roomId")]
+		public int RoomId { get; set; }
+
+		[JsonProperty(PropertyName = "createdByUserId")]
```
CreatedByUserId and CreationTimestamp won't need setters because the post request from the front-end will not be modifying these. They can be collected from the token with the post request.

mattpadgett reviewed (changes requested):
Make sure tokens are being taken in anywhere that is appropriate and that post methods aren't unnecessarily named.

Commit:
Pull Request fixes and validation

Fixed all issues outlined by matt and added token validation for all Http requests for all controllers

Commit:
http post syntax

removed bad syntax in room post request

mattpadgett reviewed (approved):
Good to go. Will merge. Thanks

mattpadgett commented:
Merged.
